runtime/lua/vim/lsp/inlay_hint.lua

```diff
@@ -4,11 +4,17 @@ local ms = require('vim.lsp.protocol').Methods
 local api = vim.api
 local M = {}
 
+---@class vim.lsp.inlay_hint.config
+---@field format_label? fun(label: string): string Formatting function for customizing an inlay hint's label
```

Comment:
What else context information (e.g. bufstate, ctx, or even the bufnr itself) could we add here?

Reply:
I think we shouldn't leak more state, since then that would tie us to keep that state around even if the implementation changes.